syntax = "proto3";

message Person {
  string name = 1;
  int32 id = 2;
  string email = 3;

  message PhoneNumber {
    enum PhoneType {
      PHONE_TYPE_UNSPECIFIED = 0;
      PHONE_TYPE_MOBILE = 1;
      PHONE_TYPE_HOME = 2;
      PHONE_TYPE_WORK = 3;
    }
  
    string number = 1;
    PhoneType type = 2;
  }

  repeated PhoneNumber phones = 4;
}

message AddressBook {
  repeated Person people = 1;
}
